### ISSUE TITLE
net.http module does not compile with `-Wall`

### ISSUE THREAD

**Issue by mcastorina** [vlang/v#8887]
**V version:** `V 0.2.2 10de905.1a838b1`
**OS:** `linux, Linux version 5.10.16-arch1-1 (linux@archlinux) (gcc (GCC) 10.2.0, GNU ld (GNU Binutils) 2.36) #1 SMP PREEMPT Sat, 13 Feb 2021 20:50:18 +0000`

**What did you do?**

```v
import net.http

fn main() {
	println(http.Method.get)
}
```

```
v -cflags '-Werror' run test.v
```

**What did you expect to see?**
```
get
```
 
**What did you see instead?**
Compilation error: assignment discards qualifiers from pointer target type

**Comment by mcastorina** [comment]
I narrowed it down to this line: https://github.com/vlang/v/blob/f18adf7759bb0cda8cd955cc5cdd32bba7ecf744/vlib/net/address.v#L39

**Comment by mcastorina** [comment]
I narrowed it down to this line:
https://github.com/vlang/v/blob/f18adf7759bb0cda8cd955cc5cdd32bba7ecf744/vlib/net/http/backend_nix.c.v#L15

**Comment by spytheman** [comment]
Try:
```v
    ssl_method := C.TLSv1_2_method()
    ctx := C.SSL_CTX_new(ssl_method)
```
to this:
```v
ctx := C.SSL_CTX_new(C.TLSv1_2_method())
```

**Comment by spytheman** [comment]
Fixed in b01a302 .
